tracer/src/Datadog.Trace/LibDatadog/CharSlice.cs

```diff
@@ -0,0 +1,50 @@
+// <copyright file="CharSlice.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Datadog.Trace.LibDatadog;
+
+/// <summary>
+/// Represents a slice of a UTF-8 encoded string in memory.
+/// </summary>
+[StructLayout(LayoutKind.Sequential)]
+internal struct CharSlice
+{
+    /// <summary>
+    /// Pointer to the start of the slice.
+    /// </summary>
+    internal IntPtr Ptr;
+
+    /// <summary>
+    /// Length of the slice.
+    /// </summary>
+    internal UIntPtr Len;
```

Comment:
```suggestion
    internal nuint Len;
```
Learned from Gregory that `nuint` is more adapted to [represent integers than pointers](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types), even though both work

Reply:
fair enough.